From 2a6968b40378dca5650e18e03ad0407738c47be5 Mon Sep 17 00:00:00 2001
From: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>
Date: Tue, 2 Jun 2026 16:36:31 -0400
Subject: [PATCH] advertising: Fix sending extra bytes with
 MGMT_OP_ADD_EXT_ADV_DATA

MGMT_OP_ADD_EXT_ADV_DATA expects the command to be of size of
struct mgmt_cp_add_ext_adv_data not mgmt_cp_add_advertising.

diff --git a/src/advertising.c b/src/advertising.c
index c8b00a887..1ed09c902 100644
--- a/src/advertising.c
+++ b/src/advertising.c
@@ -1498,8 +1498,7 @@ static void add_adv_params_callback(uint8_t status, uint16_t length,
 		}
 	}
 
-	param_len = sizeof(struct mgmt_cp_add_advertising) + adv_data_len +
-							scan_rsp_len;
+	param_len = sizeof(*cp) + adv_data_len + scan_rsp_len;
 
 	cp = malloc0(param_len);
 	if (!cp) {

